railties/lib/rails/commands/help/help_command.rb

```diff
@@ -7,6 +7,13 @@ class HelpCommand < Base # :nodoc:
 
       def help(*)
         say self.class.desc
+        if self.class.engine?
```

Comment:
This method doesn't work as expected. It seems to be buggy on master as well as I can see usage of the `new` command inside a Rails application as well when running `rails -h`.